baobab/src/baobab.c

```diff
@@ -73,13 +73,15 @@ scan_is_local (GFile	*file)
 void
 baobab_set_busy (gboolean busy)
 {
+	GdkDisplay *display;
 	static GdkCursor *busy_cursor = NULL;
 	GdkCursor *cursor = NULL;
 	GdkWindow *window;
 
 	if (busy == TRUE) {
 		if (!busy_cursor) {
-			busy_cursor = gdk_cursor_new (GDK_WATCH);
+			display = gtk_widget_get_display (GTK_WIDGET (window));
```

Comment:
```
$ mate-disk-usage-analyzer 

(mate-disk-usage-analyzer:5816): Gtk-CRITICAL **: gtk_widget_get_display: assertion 'GTK_IS_WIDGET (widget)' failed

(mate-disk-usage-analyzer:5816): Gdk-CRITICAL **: gdk_cursor_new_for_display: assertion 'GDK_IS_DISPLAY (display)' failed
```

That's because ```window``` variable isn't initialized here yet (it's done only at the end of this function).